src/runtime/hexagon/hexagon_module.h

```diff
@@ -0,0 +1,96 @@
+/*!
+ * Copyright 2018-2019 Qualcomm Innovation Center, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE‐2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * This Contribution is being provided by Qualcomm Technologies, Inc.,
+ * a Delaware corporation, or its subsidiary Qualcomm Innovation Center, Inc.,
+ * a California corporation, under certain additional terms and conditions
+ * pursuant to Section 5 of the Apache 2.0 license.  In this regard, with
+ * respect to this Contribution, the term "Work" in Section 1 of the
+ * Apache 2.0 license means only the specific subdirectory within the TVM repo
+ * (currently at https://github.com/dmlc/tvm) to which this Contribution is
+ * made.
+ * In any case, this submission is "Not a Contribution" with respect to its
+ * permitted use with any of the "vta" and "verilog" subdirectories in the TVM
+ * repo.
+ */
+#ifndef TVM_RUNTIME_HEXAGON_HEXAGON_MODULE_H_
+#define TVM_RUNTIME_HEXAGON_HEXAGON_MODULE_H_
+
+#include <dmlc/logging.h>
+#include <tvm/runtime/module.h>
+
+#include <array>
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include "../meta_data.h"
+
+namespace tvm {
+namespace runtime {
+
+/*!
+ * \brief create a Hexagon module from data.
+ *
+ * \param data The module data, can be hsaco
+ * \param fmt The format of the data, can be "obj"
+ * \param fmap The map function information map of each function.
+ */
+Module HexagonModuleCreate(std::string data, std::string fmt,
+                           std::unordered_map<std::string, FunctionInfo> fmap);
+
+namespace hexagon {
+
+class Device {
+ public:
+  virtual void* Alloc(unsigned size, unsigned align) = 0;
+  virtual void Free(void* ptr) = 0;
+  virtual void Copy(void* dst, const void* src, unsigned len) = 0;
+  virtual void CopyFrom(void* host_dst, const void* src, unsigned len) = 0;
+  virtual void CopyTo(void* dst, const void* host_src, unsigned len) = 0;
+  virtual void* Load(const std::string& data, const std::string& fmt) = 0;
```

Comment:
Can we load multiple modules? need more document

Reply:
Done.